client/src/components/providers/SingleQueryProvider.js

```diff
@@ -13,17 +22,34 @@ export const SingleQueryProvider = (cfg = {}) => {
             loading() {
                 return this.result === undefined;
             },
+            cacheKey() {
+                hash(this.$attrs || {});
+            },
         },
-        async created() {
-            if (undefined === cached_result) {
-                cached_result = await lookup();
+        mounted() {
+            const cachedResult = resultCache.get(this.cacheKey);
+            if (cachedResult) {
+                this.result = cachedResult;
```

Comment:
I don't think that makes sense for anything that can change with subsequent queries, such as the JobDetailsProvider or the JobProblemProvider. Also shouldn't cache-y things go into a vuex store ?

Reply:
Had an ongoing discussion with @Nerdinacan regarding this who drafted the initial version of the `SingleQueryProvider` afaik. The overall idea is to not store everything in vuex but still avoid duplicate/unnecessary requests. I'm open to suggestions, for now I think that using this provider is cleaner then using service/axios requests in the created function. This discussion is somewhat related to #12401, since it is the reason why we in some cases use methods instead of computed values.